crates/oxc_parser/src/lexer/mod.rs

```diff
@@ -841,7 +843,11 @@ impl<'a> Lexer<'a> {
             builder.push_matching(c);
         }
         self.read_decimal_digits(builder);
-        Kind::Float
+        if c == '-' {
```

Comment:
`~0` is `-1`. Using `-` delimiter is not sufficient to distinguish negative numbers.